src/__tests__/components/frontend-engine/frontend-engine.spec.tsx

```diff
@@ -102,8 +102,8 @@ describe("frontend-engine", () => {
 		};
 		render(<FrontendEngineWithCustomButton onClick={handleClick} />);
 
-		fireEvent.change(screen.getByTestId(fieldOneTestId), { target: { value: "hello" } });
-		fireEvent.click(screen.getByTestId(buttonTestId));
+		fireEvent.change(screen.getByRole("textbox", { name: fieldOneLabel }), { target: { value: "hello" } });
+		fireEvent.click(screen.getByRole("button", { name: customButtonId }));
```

Comment:
can help to extract these queries into local helpers e.g. getFieldOne and getCustomSubmitButton? this should improve the readability of the tests